### PR TITLE
region: update region-size from 96MB to 256MB.

### PR DESCRIPTION
**Pull request by LykxSassinator**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

This pr specify the newly updated configuration change to the default value of `region-size`, changing it from `96MB` to `256MB`.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.3 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/18336
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by LykxSassinator** [comment]
Hold it until v8.4. /cc @qiancai

File: best-practices/pd-scheduling-best-practices.md [hunk 1]
```diff
@@ -104,9 +104,9 @@ Region merge refers to the process of merging adjacent small regions. It serves
 
 Specifically, when a newly split Region exists for more than the value of [`split-merge-interval`](/pd-configuration-file.md#split-merge-interval) (`1h` by default), if the following conditions occur at the same time, this Region triggers the Region merge scheduling:
 
-- The size of this Region is smaller than the value of the [`max-merge-region-size`](/pd-configuration-file.md#max-merge-region-size) (20 MiB by default)
+- The size of this Region is smaller than the value of the [`max-merge-region-size`](/pd-configuration-file.md#max-merge-region-size) (Change to 54 MiB by default since v8.4.0, automatically applied only for newly created clusters).
```
Comment (hfxsd) [inline best-practices/pd-scheduling-best-practices.md:107]:
```suggestion
- The size of this Region is smaller than the value of the [`max-merge-region-size`](/pd-configuration-file.md#max-merge-region-size). Starting from v8.4.0, this default value is changed to 54 MiB. The new default value is automatically applied only for newly created clusters.
```

File: pd-configuration-file.md [hunk 2]
```diff
@@ -261,13 +261,13 @@ Configuration items related to scheduling
 ### `max-merge-region-size`
 
 + Controls the size limit of `Region Merge`. When the Region size is greater than the specified value, PD does not merge the Region with the adjacent Regions.
-+ Default value: `20`
++ Default value: `54` (Before v8.4.0, the default value is `20`. Starting from v8.4.0, the default value is `54`.)
```
Comment (hfxsd) [inline pd-configuration-file.md:264]:
```suggestion
+ Default value: `54`. Before v8.4.0, the default value is `20`. Starting from v8.4.0, the default value is `54`.
```

File: pd-configuration-file.md [hunk 3]
```diff
@@ -261,13 +261,13 @@ Configuration items related to scheduling
...
 + Unit: MiB
 
 ### `max-merge-region-keys`
 
 + Specifies the upper limit of the `Region Merge` key. When the Region key is greater than the specified value, the PD does not merge the Region with its adjacent Regions.
-+ Default value: `200000`
++ Default value: `540000` (Before v8.4.0, the default value is `200000`. Starting from v8.4.0, the default value is `540000`.)
```
Comment (hfxsd) [inline pd-configuration-file.md:270]:
```suggestion
+ Default value: `540000`. Before v8.4.0, the default value is `200000`. Starting from v8.4.0, the default value is `540000`.
```

File: tikv-configuration-file.md [hunk 4]
```diff
@@ -1082,7 +1082,7 @@ Configuration items related to Coprocessor.
 ### `region-split-size`
 
 + The size of the newly split Region. This value is an estimate.
-+ Default value: `"96MiB"`
++ Default value: `"256MiB"`
```
Comment (hfxsd) [inline tikv-configuration-file.md:1085]:
```suggestion
+ Default value: `"256MiB"`. Before v8.4.0, the default value is `‘96MiB’`.
```

File: tikv-configuration-file.md [hunk 5]
```diff
@@ -1093,7 +1093,7 @@ Configuration items related to Coprocessor.
 ### `region-split-keys`
 
 + The number of keys in the newly split Region. This value is an estimate.
-+ Default value: `960000`
++ Default value: `2560000`
```
Comment (hfxsd) [inline tikv-configuration-file.md:1096]:
```suggestion
+ Default value: `2560000`. Before v8.4.0, the default value is `960000’`.
```

File: tikv-configuration-file.md [hunk 6]
```diff
@@ -2147,7 +2147,7 @@ Configuration items related to BR backup.
 
 + The threshold of the backup SST file size. If the size of a backup file in a TiKV Region exceeds this threshold, the file is backed up to several files with the TiKV Region split into multiple Region ranges. Each of the files in the split Regions is the same size as `sst-max-size` (or slightly larger).
 + For example, when the size of a backup file in the Region of `[a,e)` is larger than `sst-max-size`, the file is backed up to several files with regions `[a,b)`, `[b,c)`, `[c,d)` and `[d,e)`, and the size of `[a,b)`, `[b,c)`, `[c,d)` is the same as that of `sst-max-size` (or slightly larger).
-+ Default value: `"144MiB"`
++ Default value: `"384MiB"`. The default value is `"144MiB"` in versions earlier than v8.4.0.
```
Comment (hfxsd) [inline tikv-configuration-file.md:2150]:
```suggestion
+ Default value: `"384MiB"`. Before v8.4.0, the default value is `"144MiB"`.
```

File: best-practices/pd-scheduling-best-practices.md [hunk 7]
```diff
@@ -104,9 +104,9 @@ Region merge refers to the process of merging adjacent small regions. It serves
 
 Specifically, when a newly split Region exists for more than the value of [`split-merge-interval`](/pd-configuration-file.md#split-merge-interval) (`1h` by default), if the following conditions occur at the same time, this Region triggers the Region merge scheduling:
 
-- The size of this Region is smaller than the value of the [`max-merge-region-size`](/pd-configuration-file.md#max-merge-region-size) (20 MiB by default)
+- The size of this Region is smaller than the value of the [`max-merge-region-size`](/pd-configuration-file.md#max-merge-region-size). Starting from v8.4.0, the default value is changed to 54 MiB. The new default value is automatically applied only for newly created clusters. Existing clusters are not affected.
```
Comment (hfxsd) [inline best-practices/pd-scheduling-best-practices.md:107]:
```suggestion
- The size of this Region is smaller than the value of the [`max-merge-region-size`](/pd-configuration-file.md#max-merge-region-size). Starting from v8.4.0, the default value is changed from 20 MiB to 54 MiB. The new default value is automatically applied only for newly created clusters. Existing clusters are not affected.
```

File: best-practices/massive-regions-best-practices.md [hunk 8]
```diff
@@ -138,7 +138,11 @@ If Region followers have not received the heartbeat from the leader within the `
 
 ### Method 6: Adjust Region size
 
-The default size of a Region is 96 MiB, and you can reduce the number of Regions by setting Regions to a larger size. For more information, see [Tune Region Performance](/tune-region-performance.md).
+The default size of a Region is 256 MiB, and you can reduce the number of Regions by setting Regions to a larger size. For more information, see [Tune Region Performance](/tune-region-performance.md).
+
+> **Note:**
+>
+> Starting from v8.4.0, the default size of Region is changed to 256 MiB.
```
Comment (hfxsd) [inline best-practices/massive-regions-best-practices.md:145]:
```suggestion
> Starting from v8.4.0, the default Region size has been increased to 256 MiB. When upgrading a cluster to v8.4.0 or later, if you do not adjust the Region size manually, the TiKV cluster's default Region size will automatically update to 256 MiB.
```

**Review by songrijie — APPROVED** [review]

File: best-practices/massive-regions-best-practices.md [hunk 9]
```diff
@@ -138,7 +138,11 @@ If Region followers have not received the heartbeat from the leader within the `
 
 ### Method 6: Adjust Region size
 
-The default size of a Region is 96 MiB, and you can reduce the number of Regions by setting Regions to a larger size. For more information, see [Tune Region Performance](/tune-region-performance.md).
+The default size of a Region is 256 MiB, and you can reduce the number of Regions by setting Regions to a larger size. For more information, see [Tune Region Performance](/tune-region-performance.md).
+
+> **Note:**
+>
+> Starting from v8.4.0, the default Region size has been increased to 256 MiB. When upgrading a cluster to v8.4.0 or later, if the Region size has not been changed manually, the TiKV cluster's default Region size will automatically update to 256 MiB.
```
Comment (lilin90) [inline best-practices/massive-regions-best-practices.md:145]:
```suggestion
> Starting from v8.4.0, the default Region size is increased from 96 MiB to 256 MiB. If you have not modified the Region size manually, when you upgrade a TiKV cluster to v8.4.0 or later,  the TiKV cluster's default Region size will automatically be updated to 256 MiB.
```

File: best-practices/pd-scheduling-best-practices.md [hunk 10]
```diff
@@ -104,9 +104,9 @@ Region merge refers to the process of merging adjacent small regions. It serves
 
 Specifically, when a newly split Region exists for more than the value of [`split-merge-interval`](/pd-configuration-file.md#split-merge-interval) (`1h` by default), if the following conditions occur at the same time, this Region triggers the Region merge scheduling:
 
-- The size of this Region is smaller than the value of the [`max-merge-region-size`](/pd-configuration-file.md#max-merge-region-size) (20 MiB by default)
+- The size of this Region is smaller than the value of the [`max-merge-region-size`](/pd-configuration-file.md#max-merge-region-size). Starting from v8.4.0, the default value is changed from 20 MiB to 54 MiB. The new default value is automatically applied only for newly created clusters. Existing clusters are not affected.
```
Comment (lilin90) [inline best-practices/pd-scheduling-best-practices.md:107]:
```suggestion
- The size of this Region is smaller than the value of the [`max-merge-region-size`](/pd-configuration-file.md#max-merge-region-size). Starting from v8.4.0, the default value is changed from 20 MiB to 54 MiB. The new default value is automatically applied only to newly created clusters. Existing clusters are not affected.
```

File: best-practices/pd-scheduling-best-practices.md [hunk 11]
```diff
@@ -104,9 +104,9 @@ Region merge refers to the process of merging adjacent small regions. It serves
...
 
-- The number of keys in this Region is smaller than the value of [`max-merge-region-keys`](/pd-configuration-file.md#max-merge-region-keys) (200,000 by default).
+- The number of keys in this Region is smaller than the value of [`max-merge-region-keys`](/pd-configuration-file.md#max-merge-region-keys). Starting from v8.4.0, the default value is changed from 200000 to 540000. The new default value is automatically applied only for newly created clusters. Existing clusters are not affected.
```
Comment (lilin90) [inline best-practices/pd-scheduling-best-practices.md:109]:
```suggestion
- The number of keys in this Region is smaller than the value of [`max-merge-region-keys`](/pd-configuration-file.md#max-merge-region-keys). Starting from v8.4.0, the default value is changed from 200000 to 540000. The new default value is automatically applied only to newly created clusters. Existing clusters are not affected.
```

File: tikv-configuration-file.md [hunk 12]
```diff
@@ -1082,7 +1082,7 @@ Configuration items related to Coprocessor.
 ### `region-split-size`
 
 + The size of the newly split Region. This value is an estimate.
-+ Default value: `"96MiB"`
++ Default value: `"256MiB"`. Before v8.4.0, the default value is `‘96MiB’`.
```
Comment (lilin90) [inline tikv-configuration-file.md:1085]:
```suggestion
+ Default value: `"256MiB"`. Before v8.4.0, the default value is `"96MiB"`.
```

File: tikv-configuration-file.md [hunk 13]
```diff
@@ -1093,7 +1093,7 @@ Configuration items related to Coprocessor.
 ### `region-split-keys`
 
 + The number of keys in the newly split Region. This value is an estimate.
-+ Default value: `960000`
++ Default value: `2560000`. Before v8.4.0, the default value is `960000’`.
```
Comment (lilin90) [inline tikv-configuration-file.md:1096]:
```suggestion
+ Default value: `2560000`. Before v8.4.0, the default value is `960000`.
```

File: tune-tikv-memory-performance.md [hunk 14]
```diff
@@ -111,8 +111,8 @@ region-split-check-diff = "32MB"
 [coprocessor]
 ## If the size of a Region with the range of [a,e) is larger than the value of `region_max_size`, TiKV tries to split the Region to several Regions, for example, the Regions with the ranges of [a,b), [b,c), [c,d), and [d,e).
 ## After the Region split, the size of the split Regions is equal to the value of `region_split_size` (or slightly larger than the value of `region_split_size`).
-# region-max-size = "144MB"
-# region-split-size = "96MB"
+# region-max-size = "384MB"
+# region-split-size = "256MB"
```
Comment (lilin90) [inline tune-tikv-memory-performance.md:115]:
```suggestion
# region-max-size = "384MiB"
# region-split-size = "256MiB"
```

**Review by lilin90 — APPROVED** [review]
/lgtm

**Review by hfxsd — APPROVED** [review]

**Pull request merged**